### DISCUSSION TITLE
Faulting module name: coreclr.dll

### DISCUSSION THREAD

**Discussion by LukAss741** [dotnet/winforms#7758] [Q&A] [answered]
### .NET version

new6.0-windows

### Did it work in .NET Framework?

No

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

6.0.422.16404 didn't work either.

### Issue description

My application that is supposed to run 24/7 regularly crashes. This is what I see in event viewer:

```
Faulting application name: xx.exe, version: 1.0.0.0, time stamp: 0x62cf0990
Faulting module name: coreclr.dll, version: 6.0.822.36306, time stamp: 0x62cf0045
Exception code: 0xc0000005
Fault offset: 0x00000000001d1fdd
Faulting process id: 0x3ed8
Faulting application start time: 0x01d8c47bc2e2313a
Faulting application path: C:\Data\xx.exe
Faulting module path: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\6.0.8\coreclr.dll
Report Id: 525e6274-83f7-4540-8f3a-471a6655e70a
Faulting package full name: 
Faulting package-relative application ID: 
```

### Steps to reproduce

I have a large project. The program crashes very iregularly - sometimes every like 10 mintes other times it may run for a week without a single crash. Hence, I don't know what exactly causes the crash. Can you advise how to proceed?

**Comment by kirsan31** [comment, marked as answer]
You need to investigate crush dump. I suggest to create a full dump. Look here: https://docs.microsoft.com/en-us/windows/win32/wer/collecting-user-mode-dumps

**Reply by LukAss741** [reply]
Thanks. I made the change according to the link. Now I am waiting for the software to crash. Surprisingly it has been running without a single crash since then.

**Comment by LukAss741** [comment]
From the dump I discovered this was probably caused by improper disposal of an object. Thank you for pointing me the right direction.
Anyway is it normal that it didn't get caught but only crached the process with access violation exception?

**Reply by RussKie** [reply]
IIUIC, AVE (exception code: 0xc0000005) are fatal, and the OS terminates the process that caused it. But your best bet to get a better answer is to ask the question the area owners in [dotnet/runtime](https://github.com/dotnet/runtime).

**Reply by kirsan31** [reply]
https://github.com/dotnet/runtime/issues/13805
